hack/drone.jsonnet

```diff
@@ -0,0 +1,323 @@
+    "git checkout ${DRONE_SOURCE_BRANCH}",
+    "git fetch --tags",
+  ],
+  when: {
+    event: {
+      exclude: [""],
+    },
+  },
+};
+
+local docker = {
+  name: "docker",
+  image: dind_image,
+  entrypoint: ["dockerd"],
+  privileged: true,
+  command: [
+    "--dns=8.8.8.8",
+    "--dns=8.8.4.4",
+    "--mtu=1440",
+    "--log-level=error",
+  ],
+  volumes: step_volumes,
+};
+
+local buildkit = {
+  name: "buildkit",
+  image: "moby/buildkit:v0.6.0",
+  privileged: true,
+  detach: true,
+  commands: ["buildkitd --addr tcp://0.0.0.0:1234 --allow-insecure-entitlement security.insecure"],
+  when: {
+    event: {
+      include: [""],
+    },
+  },
+};
+
+// local push = {
+//   name: "push",
+//   image: "autonomy/build-container:latest",
+//   pull: "always",
+//   environment:
+//     DOCKER_USERNAME:
+//       from_secret: docker_username
+//     DOCKER_PASSWORD:
+//       from_secret: docker_password
+//   commands:
+//     - make gitmeta
+//     - make login
+//     - make push
+//   volumes:
+//     - name: dockersock
+//       path: /var/run
+//   when:
+//     event:
+//       - push
+//   depends_on:
+//     - test.Integration
+// };
+
+local notify = {
+  name: "slack",
+  image: "plugins/slack",
+  settings:
+    {
+     webhook: {from_secret: "slack_webhook"},
+     channel: "proj-talos-maint",
+    },
+};
+
+// Functions
+
+local step(name, depends_on=[], environment={}, target="") = {
+  name: std.format("%s", name),
+  image: build_container,
+  commands: [if target == "" then std.format("make %s", name) else std.format("make %s", target)],
+  environment: build_env_vars+environment,
+  volumes: step_volumes,
+  depends_on: [x.name for x in depends_on],
+};
+
+
+local pipeline(name, steps=[]) = {
+  kind: "pipeline",
+  name: name,
+  node: node,
+  services: if name != "notify" then [docker] else [],
+  volumes: if name != "notify" then pipeline_volumes else [],
+  clone: {
+    disable: true,
+  },
+  steps: if name != "notify" then [clone, buildkit] + steps else steps,
+};
+
+// Apps
+
+local machined = step("machined", [clone]);
+local osd = step("osd", [clone]);
+local trustd = step("trustd", [clone]);
+local proxyd = step("proxyd", [clone]);
+local ntpd = step("ntpd", [clone]);
+local osctl_linux = step("osctl-linux", [clone]);
+local osctl_darwin = step("osctl-darwin", [clone]);
+
+// Artifacts
+
+local rootfs =  step("rootfs", [machined, osd, trustd, proxyd, ntpd]);
+local initramfs = step("initramfs", [rootfs]);
+local installer = step("installer", [rootfs]);
+local container = step("container", [rootfs]);
+local image_azure = step("image-azure", [installer]);
+local image_gce = step("image-gce", [installer]);
+local kernel = step("kernel", [clone]);
+local iso = step("iso", [installer]);
+
+
+// CAPI.
+
+local push_image_azure = step("push-image-azure", [image_azure], creds_env_vars);
+local push_image_gce = step("push-image-gce", [image_gce], creds_env_vars);
+
+// Tests
+
+local lint = step("lint", [rootfs]);
```

Comment:
why move lint to depend on rootfs and not clone?

Reply:
I was thinking that we could group all the "tests" to run at the same time, but its probably better to just depend on `clone`. I will update.